packages/dd-trace/src/ci-visibility/exporters/agentless/index.js

```diff
@@ -2,18 +2,51 @@
 
 const URL = require('url').URL
 const Writer = require('./writer')
+const CoverageWriter = require('./coverage-writer')
 const Scheduler = require('../../../exporters/scheduler')
 
+const log = require('../../../log')
+
 class AgentlessCiVisibilityExporter {
   constructor (config) {
-    const { flushInterval, tags, site, url } = config
+    const { flushInterval, tags, site, url, isIntelligentTestRunnerEnabled } = config
+    this._isIntelligentTestRunnerEnabled = isIntelligentTestRunnerEnabled
+
     this._url = url || new URL(`https://citestcycle-intake.${site}`)
     this._writer = new Writer({ url: this._url, tags })
 
+    this._coverageUrl = new URL(`https://event-platform-intake.${site}`)
+    this._coverageWriter = new CoverageWriter({ url: this._coverageUrl })
+
     if (flushInterval > 0) {
       this._scheduler = new Scheduler(() => this._writer.flush(), flushInterval)
+
+      if (this._isIntelligentTestRunnerEnabled) {
+        this._coverageScheduler = new Scheduler(() => this._coverageWriter.flush())
+      }
     }
+
     this._scheduler && this._scheduler.start()
+
+    // Reduce likelihood of requests overlapping
```

Comment:
Why is this needed?

Reply:
I figured it'd be better if we don't fire two requests at the same time. I can remove it if you think it doesn't add much. In any case, after we merge https://github.com/DataDog/dd-trace-js/pull/2296 this won't look like this